paddle/fluid/operators/mkldnn/conv_transpose_mkldnn_op.cc

```diff
@@ -229,32 +229,21 @@ class ConvTransposeMKLDNNHandlerT
         weights_tz, platform::MKLDNNGetDataType<K>(),
         (g == 1) ? filter->format() : MKLDNNMemoryFormat::goihw);
 
-    auto iohw_weights_tz = framework::vectorize(filter->dims());
-    // Custom Reorder from IOHW to OIHW
-    auto iohw2oihw_reorder =
-        [&iohw_weights_tz](const K* filter_data) -> std::shared_ptr<K> {
-      int o = iohw_weights_tz[1];
-      int c = iohw_weights_tz[0];
-      int h = iohw_weights_tz[2];
-      int w = iohw_weights_tz[3];
-      std::shared_ptr<K> reordered_filter_data(new K[o * c * h * w](),
-                                               std::default_delete<K[]>());
-      for (int i = 0; i < c; ++i) {
-        for (int j = 0; j < o; ++j) {
-          int in_offset = j * h * w + i * o * h * w;
-          int out_offset = j * c * h * w + i * h * w;
-          std::memcpy(&(reordered_filter_data.get())[out_offset],
-                      &filter_data[in_offset], h * w * sizeof(K));
-        }
-      }
-
-      return reordered_filter_data;
-    };
+    const platform::MKLDNNEngine& engine = dev_ctx.GetEngine();
```

Comment:
I actually think  this block of code is not needed. What should be modified is fragment "filter->format()" to "dnnl::format_tag::iohw" (guess) and then AcquireMemoryWithReorder will take care of rordering from IOHW -> blocked_optimized format. In your current changes you have two reorders: IOHW -> OIHW and then OIHW to blocked_optimized_format.  One issue here is that filter-<format() is returning (probably) OIHW while in fact data is IOHW .